File: clients/web/src/api/shares.api.ts
import type { CreateShareRequest, CreateShareResponse, ShareDetailsResponse, ShareLinkOwnerResponse, ShareLinkResponse } from '../types/share';
import { api } from './client';

export const sharesApi = {
  create: (folderId: string, data: CreateShareRequest) =>
    api.post<CreateShareResponse>(`/api/sharing/folders/${folderId}/share`, data),

  list: (folderId: string) =>
    api.get<ShareLinkResponse[]>(`/api/sharing/folders/${folderId}/shares`),

  revokeById: (id: string) =>
    api.delete(`/api/sharing/share-links/${id}`),

  revokeByToken: (token: string) =>
    api.delete(`/api/sharing/share/${token}`),

  getByToken: (token: string) =>
    api.get<ShareDetailsResponse>(`/api/sharing/share/${token}`),

  listAll: () =>
    api.get<ShareLinkOwnerResponse[]>('/api/sharing/share-links'),
};

File: clients/desktop/GraNAS.Desktop.App/Services/Api/SharesApi.cs
using GraNAS.Desktop.Contracts.Sharing;

namespace GraNAS.Desktop.App.Services.Api;

public class SharesApi : ApiBase, ISharesApi
{
  public SharesApi(HttpClient http) : base(http) { }

  public Task<List<ShareLinkResponse>> GetSharesAsync(Guid folderId, CancellationToken ct = default)
    => GetAsync<List<ShareLinkResponse>>($"api/sharing/folders/{folderId}/shares", ct);

  public Task<CreateShareResponse> CreateShareAsync(Guid folderId, CreateShareRequest request, CancellationToken ct = default)
    => PostAsync<CreateShareResponse>($"api/sharing/folders/{folderId}/share", request, ct);

  public Task RevokeShareAsync(Guid shareLinkId, CancellationToken ct = default)
    => DeleteAsync($"api/sharing/share-links/{shareLinkId}", ct);

  public Task<ShareDetailsResponse> GetShareDetailsAsync(string token, CancellationToken ct = default)
    => GetAsync<ShareDetailsResponse>($"api/sharing/share/{token}", ct);

  public async Task<List<ShareLinkOwnerResponse>> ListAllSharesAsync(bool activeOnly = true, CancellationToken ct = default)
  {
    try { return await GetAsync<List<ShareLinkOwnerResponse>>($"api/share-links?activeOnly={activeOnly}", ct); }
    catch { return []; }
  }
}
